src/Microsoft.ML.LightGBM/LightGbmBinaryTrainer.cs

```diff
@@ -111,20 +111,15 @@ internal LightGbmBinaryTrainer(IHostEnvironment env, LightGbmArguments args)
         /// <param name="numBoostRound">Number of iterations.</param>
         /// <param name="minDataPerLeaf">The minimal number of documents allowed in a leaf of the tree, out of the subsampled data.</param>
         /// <param name="learningRate">The learning rate.</param>
-        /// <param name="advancedSettings">A delegate to set more settings.
-        /// The settings here will override the ones provided in the direct signature,
-        /// if both are present and have different values.
-        /// The columns names, however need to be provided directly, not through the <paramref name="advancedSettings"/>.</param>
-        public LightGbmBinaryTrainer(IHostEnvironment env,
+        internal LightGbmBinaryTrainer(IHostEnvironment env,
             string labelColumn = DefaultColumnNames.Label,
```

Comment:
Is it not possible to combine the two constructors?
Or have you added this work to the reconciliation issue #2100?

Reply:
yeap. exactly.  added to #2100

want to focus on cleanup of public surface first

---
In reply to: [251541952](https://github.com/dotnet/machinelearning/pull/2261#discussion_r251541952) [](ancestors = 251541952)